src/lexitools/commands/correspondences.py

```diff
@@ -1,161 +1,373 @@
+
+    Attributes:
+        datasets (dict of str to cldfbench.Dataset): maps dataset names to Dataset objects.
+    """
 
     def __init__(self, dataset_list):
+        """ Load all datasets from a list of dataset indentifiers, or install them.
+
+        Note: this does NOT update installed datasets which are out of date.
+
+        Args:
+            dataset_list (list): list of dataset identifiers. Each identifier is a 2-tuple
+                of a github organization name ("lexibank" or "sequencecomparison" in most
+                cases) and a lexibank git repository name (such as "cals").
+        """
         # see https://github.com/chrzyki/snippets/blob/main/lexibank/install_datasets.py
-        egg_pattern = "git+https://github.com/lexibank/{name}.git#egg=lexibank_{name}"
-        for name in dataset_list:
+        egg_pattern = "git+https://github.com/{org}/{name}.git#egg=lexibank_{name}"
+        needed_install = False
+        for org, name in dataset_list:
             if find_spec("lexibank_" + name) is None:
-                subprocess.run(
-                    ["pip", "install", "-e", egg_pattern.format(name=name)])
+                args = [sys.executable, "-m", "pip", "install",
+                        "-e", egg_pattern.format(org=org, name=name)]
+                subprocess.run(args)
+                needed_install = True
+
+        if needed_install:
+            raise EnvironmentError("Some datasets were not installed. ",
+                                   "I have tried to install them,",
+                                   "please re-run the command now.")
+
         self.datasets = {}
-        for name in dataset_list:
+        for org, name in dataset_list:
             try:
                 self.datasets[name] = get_dataset(name,
                                                   ep="lexibank.dataset").cldf_reader()
             except AttributeError:
                 print("Failed to load", name)
+        if not self.datasets:
+            raise Exception("No dataset loaded")
 
     def iter_table(self, table_name):
+        """ Iter on the rows of a specific table, across all loaded datasets.
+
+        Args:
+            table_name (str): name of a CLDF Wordlist table.
+
+        Yields:
+            rows from all datasets, with an additional "dataset" field indicating the name
+            of the dataset from which the row originates.
+        """
         for name, ds in self.datasets.items():
             for row in ds[table_name]:
                 row["dataset"] = name
                 yield row
 
     def get_table(self, table_name):
+        """ Return all the rows of a specific tablee, across all loaded datasets.
+
+        Args:
+            table_name (str): name of a CLDF Wordlist table.
+
+        Returns:
+            list of rows
+        """
         return list(self.iter_table(table_name))
 
 
-class SoundCorrespsByGenera(object):
+@dataclass(eq=True, frozen=True)
```

Comment:
nice to see dataclasses in use!

Reply:
I spotted them in some lexibank code somewhere and wanted to try them out instead of my usual NamedTuples. So far, I like them :)